staging/src/k8s.io/client-go/applyconfigurations/autoscaling/v1/horizontalpodautoscalerstatus.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by applyconfiguration-gen. DO NOT EDIT.
+
+package v1
+
+import (
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HorizontalPodAutoscalerStatusApplyConfiguration represents an declarative configuration of the HorizontalPodAutoscalerStatus type for use
+// with apply.
+type HorizontalPodAutoscalerStatusApplyConfiguration struct {
+	ObservedGeneration              *int64   `json:"observedGeneration,omitempty"`
+	LastScaleTime                   *v1.Time `json:"lastScaleTime,omitempty"`
+	CurrentReplicas                 *int32   `json:"currentReplicas,omitempty"`
+	DesiredReplicas                 *int32   `json:"desiredReplicas,omitempty"`
+	CurrentCPUUtilizationPercentage *int32   `json:"currentCPUUtilizationPercentage,omitempty"`
+}
+
+// HorizontalPodAutoscalerStatusApplyConfiguration constructs an declarative configuration of the HorizontalPodAutoscalerStatus type for use with
+// apply.
+func HorizontalPodAutoscalerStatus() *HorizontalPodAutoscalerStatusApplyConfiguration {
+	return &HorizontalPodAutoscalerStatusApplyConfiguration{}
+}
+
+// WithObservedGeneration sets the ObservedGeneration field in the declarative configuration to the given value
+// and returns the receiver, so that objects can be build by chaining "With" function invocations.
+func (b *HorizontalPodAutoscalerStatusApplyConfiguration) WithObservedGeneration(value int64) *HorizontalPodAutoscalerStatusApplyConfiguration {
+	b.ObservedGeneration = &value
+	return b
+}
+
+// WithLastScaleTime sets the LastScaleTime field in the declarative configuration to the given value
+// and returns the receiver, so that objects can be build by chaining "With" function invocations.
+func (b *HorizontalPodAutoscalerStatusApplyConfiguration) WithLastScaleTime(value v1.Time) *HorizontalPodAutoscalerStatusApplyConfiguration {
```

Comment:
Maybe when go2 is here we can let users optionally instead pass in a real time.Time, this looks a little annoying to use.

Reply:
I agree. We're using the same type that is used on the `HorizontalPodAutoscalerStatus`, so we didn't make it any worse (generating a TimeApplyConfiguration would have been much worse). `time.Time` would be much nicer.

Reply:
I almost suggested just using a time.Time, but then what if people want to explicitly manage the rounding? Hm maybe that's not that important. What bad thing happens if we auto-convert time.Time to v1.Time / v1.MicroTime as needed?

Reply:
If we did try to leverage `time.Time` and avoid the conversions we still wouldn't be able to change the underlying struct use `time.Time` because it doesn't support json serialization. So we'd have `time.Time` as the argument for `WithTimeField()` functions but the `TimeField` field would be `v1.Time`, and developers would be able to see that.

The k/k codebase is full of `metav1.NewTime()` calls to convert `time.Time` `metav1.Time` so I'm not sure this a huge net win given that everything else would still require `NewTime()` conversions.

MicroTime looks like the same situation since the `NewMicroTime()` function takes a `time.Time` and already handles the conversion.

Reply:
OK, sounds good.